upgrading.md

```diff
@@ -0,0 +1,53 @@
+# Upgrading
+
+This document will outline the steps to upgrade between stable releases of ManageIQ in OpenShift
+
+## Upgrading to Hammer
+
+This upgrade will utilize the built-in capabilities of the `oc apply` command to patch existing objects and deploy new images. The high level steps look like this:
+
+1. Retrieve Secrets
+2. Make Application Changes
+3. Patch OpenShift Objects
+
+### Retrieve Secrets
+
+During template resolution, secrets such as the admin password, encryption key, and database password will be re-generated if they are not provided. Because of this we need to ensure we bring forward the same values which were set up in the previous version.
+
+To retrieve the secret object run `oc export secret manageiq-secrets > my_secrets.yml`. We will overlay these values on top of the generated ones after the upgrade.
+
+### Make Application Changes
+
+Note: If you have *never* enabled the Embedded Ansible role, then you can skip these steps and the secrets will be created for you if you choose to use the role in the future.
+
+If the Embedded Ansible feature is or was in use the passwords in the ansible secret need to be moved into the application for the new version.
+Fetch the values using the command `oc export secret ansible-secrets`. There should be 3 Base64 encoded values in the secret. Substitute the values for the placeholders in angle brackets (<>) in the commands below.
+
+Apply the values to the application by accessing a manageiq pod (`oc rsh manageiq-0`, for example) and running the following commands in the rails console (`bash; vmdb; bin/rails c`)
+
+```ruby
+> db = MiqDatabase.first
+> db.ansible_secret_key = Base64.decode64("<secret-key>")
```

Comment:
Can this live in a data migration?

Reply:
That's a really good question. We would have to utilize `KubeClient` to talk to the OpenShift API during the migration to fetch the values, but if that's okay it might be possible (and definitely would be better). I'll investigate that option.